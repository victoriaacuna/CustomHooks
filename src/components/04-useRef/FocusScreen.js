import React, { useRef } from 'react'

export const FocusScreen = () => {

    // El useRef puede cambiar a lo que apunta el ref sin tener que reenderizar.
    const inputRef = useRef()

    const handleCLick = () => {
        inputRef.current.select();
        console.log(inputRef);
    }


    return (
        <div>
            <h1>Focus Screen</h1>

            <input 
                className="form-control"
                placeholder="Su nombre..."
                ref={inputRef}
            />

            <button 
                className="btn btn-outline-primary mt-5"
                onClick={handleCLick}
            >
                Focus
            </button>
        </div>
    )
}
