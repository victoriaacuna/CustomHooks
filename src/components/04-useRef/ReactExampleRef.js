import React, { useState } from 'react';
import {MultipleCustomHooks} from '../03-examples/MultipleCustomHooks';

export const ReactExampleRef = () => {

    const [show, setShow] = useState(false)

    return (
        <div>
            <h1>React Example Ref</h1>
            {
                show 
                &&
                (
                    <MultipleCustomHooks/>
                )
            }
            <button 
                className="btn btn-primary" 
                onClick={() => {setShow(!show)}}>
                Show/Hide
            </button>
            
        </div>
    )
}
