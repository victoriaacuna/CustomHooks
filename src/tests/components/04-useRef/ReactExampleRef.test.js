import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import { ReactExampleRef } from '../../../components/04-useRef/ReactExampleRef';

describe('Pruebas en React Example Ref', () => {
    
    const wrapper = shallow(<ReactExampleRef/>);

    test('should match snapshot', () => {
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBeFalsy();
    });

    test('should show <MultipleCustomHooks/>', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBeTruthy();
    });
    
    


})
