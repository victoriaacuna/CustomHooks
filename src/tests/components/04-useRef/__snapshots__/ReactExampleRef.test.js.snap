// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en React Example Ref should match snapshot 1`] = `
<div>
  <h1>
    React Example Ref
  </h1>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
